source/includes/_security.md

```diff
@@ -1,5 +1,132 @@
+
+The CDS Service is able to use the FHIR server of the EHR to obtain any additional data it needs in order to perform its decision support. This is similar to SMART on FHIR where the SMART app can obtain additional data via the provided FHIR server.
+
+Like SMART on FHIR, CDS Hooks requires that access to the FHIR server be controlled by an Authorization server utilizing the OAuth 2 framework. Thus, the CDS Service is able to consume the given FHIR server via an access (bearer) token just like a SMART app. While CDS Hooks shares the underlying technical framework and standards as SMART on FHIR, there are very important differences between SMART and CDS Hooks.
+
+### Obtaining an Access Token
+
+In SMART on FHIR, the SMART app requests and ultimately obtains an access token from the Authorization server using the SMART launch workflow. This process utilizes the authorization code grant model as defined by the OAuth 2.0 Authorization Framework in [rfc6749](https://tools.ietf.org/html/rfc6749).
+
+With CDS Hooks, the EHR provides the access token directly in the request to the CDS Service. Thus, the CDS Service does not need to request the token from the authorization server as a SMART app would. This is done purely for performance reasons as the authorization code grant model in OAuth 2 involves several HTTPS calls and redirects. In contrast with a SMART app, a CDS Service may be invoked many times during a workflow. Going through the authorization code grant model on every hook invocation would likely result in a slow performing CDS Service due to the authorization overhead.
+
+```json
+{
+  "fhirAuthorization" : {
+    "access_token" : "some-opaque-fhir-access-token",
+    "token_type" : "code",
+    "expires_in" : 300,
+    "scope" : "patient/Patient.read patient/Observation.read"
+  }
+}
+```
+
+### Access Token
+
+The access token is specified in the CDS Service request via the `fhirAuthorization` request parameter. This parameter is an object that contains both the access token as well as other related information.
+
+Field | Description
+----- | -----------
+`access_token` |*string*. This is the OAuth 2 access token that provides access to the FHIR server.
+`token_type`   |*string*. Always the value `code`
+`expires_in`   |*integer*. The lifetime in seconds of the access token.
+`scope`        |*string*. The scopes the access token grants the CDS Service.
+
+It is recommended that the `expires_in` value be very short lived as the access token should be treated as a transient value by the CDS Service.
+
+It is recommended that the `scope` value contain just the scopes that the CDS Service needs for its logic and no more.
+
+As the CDS Service is executing on behalf of a user, it is important that the data the CDS Service has access to is under the same restrictions/authorization as the current user. As such, the access token should be scoped to:
+
+- The CDS Service being invoked
+- The current user
+- The current patient
```

Comment:
Many EHRs are multi-tenant and may need the token to be scoped by a practice/organization.  Some organization models also allow users to have access to multiple organizations so it is not safe to assume that the current user maps one-to-one with an organization or practice.

Reply:
I'm not assuming the user maps one-to-one with and org or practice. The point I was trying to make here is that the access token needs to be narrowly scoped such that if it it cannot be misused by the CDS Service to obtain data the current user cannot see. This line echoes that point:

> As the CDS Service is executing on behalf of a user, it is important that the data the CDS Service has access to is under the same restrictions/authorization as the current user.

Is there a way you can suggest that I make this more clear?

(As an aside, Cerner's EHR can be run in a single or multi-tenant mode so I am always thinking of both models to make sure whatever we do works for both)

Reply:
It might be necessary to optionally pass additional context (i.e, add an `organization_id`) to clearly identify which organization owns the data.

Reply:
Do you have a SMART app today? If so, how are you handling this today?

Reply:
We haven't yet solved that problem, although it's going to become an issue for us soon.